### DISCUSSION TITLE
Feature request

### DISCUSSION THREAD

**Discussion by JUser69** [jessedp/tablo-tools-electron#145] [Ideas]
How hard would it be to add an export / import feature for documenting / changing recording passes (in tivo speak "one passes")?
Kinda like kmttg.jar does.  Series, channels, date/times ...

see https://sourceforge.net/p/kmttg/wiki/Home/

**Comment by jessedp** [comment]
I'm not really sure what you're asking for here, though am not familiar with the details of how Tivos function. Are you trying to transfer recording schedules between Tablo devices? Can you explain in more detail what actual functionality you're looking for, or maybe  to specific parts of kmttg's documentation that explains that?

**Reply by JUser69** [reply]
A screen that would display that information in a tabular format and that could be exported to a .csv file type.  This could be used in case one has to replace their tablo or hard drive to rebuild schedules (not recover data).  See below

BTW: the application is quiet nice as it is.  Perhaps under settings you could add a tools section where there could be things like scan disk for errors, defrag disk, etc.  I dont know what level of access to the underlying OS you have.    The tablo itself is lacking of any tools at all.  There's not even a reboot option.





![image](https://user-images.githubusercontent.com/42041388/200008809-66d2c755-fde5-48b2-af00-7f159b25e982.png)
[Living Room.csv](https://github.com/jessedp/tablo-tools-electron/files/9939633/Living.Room.csv)

**Reply by dougK44883** [reply]
Tablo Tools it awesome... for a "tabled" view look to APL Tablo - https://community.tablotv.com/t/apl-tablo-client/20222  Lots of sorting options and views as well as cvs exports.  Includes features Tablo Tools doesn't have, scheduling... it's not a GUI, while Tablo Tools has features APL Tablo does not, watch live TV.

How ever, the schedule is not stored on the drive, though there is a backup DB.  So exporting it - you can't directly import it back.  There is a backup script https://community.tablotv.com/t/backup-tablo-sh-backup-all-recorded-shows-from-a-tablo/21267 which will also save the DB directory.

As for disk checking, tablo has posted of course users don't get root access to their devices https://community.tablotv.com/t/hard-drive-clone/2585/5  But since it's not a Windows based system, defrag is not a notorious issue.  

The reboot option gets brought up often... for otherwise single user or _head-of-household_ devices.  If you have have a family and watching or recording and someone thinks they know what they're doing an clicks reboot -- so now everyone wonders why they can't watch TV or what happened to their show.  And again, not Windows based so it doesn't need routine reboots for no specific reason.

Tablo does run overnight maintenance mode - https://support.tablotv.com/hc/en-us/articles/115002622263-What-is-maintenance-mode- which reportedly tends whatever tools you feel are missing.  Non-Windows OSs don't need to constant manual intervention.

**Reply by JUser69** [reply]
All very good information and useful, thanks for the links.  All though not directly addressing my feature request, it does answer the main question that it would require to much effort to incorporate into your interface.  I understand and do appreciate your responses to my questions on the subject.  Once again, very good program.   Thank you.

**Reply by jessedp** [reply]
Currently I don't pull the full schedule down (and haven't tried). If I do that at some point, I'll certainly consider something like this as it would be rather trivial.

What you're currently seeing in the app is just a list of the current Recordings that reside on the Tablo, occasionally grouped by type or associated series/show. That leaves a lot of room for holes if I exported it and claimed it does what you're looking for.

As far as the tools mentioned, @dougK44883 pretty much covered the options above (basically, none). 

If you're comfortable hooking up the hard drive to a computer and messing around, you may find the [comments here](https://github.com/jessedp/tablo-tools-electron/issues/96#issuecomment-852273373) and Doug's just below them useful, particularly accessing `db/Tablo.db` directly - all of the schedule information will be in there along with the rest of the data the Tablo stores. Be careful since *changing* things could end up forcing your Tablo to want/need to reformat the drive.

**Reply by JUser69** [reply]
While on the subject, something that I think people would find useful would be to further define the parameters before a download.  IE., I have to re-encode everything I am going to archive.  I personally use cbr26 @ 96kbs, which works for me.  The ability to set the bitrates and sizing  for video / audio would save a lot of time. Example: College football I re-encode to 30cbr @ 96kbs x 576p.

**Reply by jessedp** [reply]
> Currently I don't pull the full schedule down (and haven't tried). If I do that at some point, I'll certainly consider something like this as it would be rather trivial.

I'm playing around generating something like [this](https://github.com/jessedp/tablo-tools-electron/files/10808778/schedule-export.csv)  - obviously the data available from a Tablo is different from the tivo. 

Let me know what you think is missing, if you have questions about that, etc.

**Reply by jessedp** [reply]
> While on the subject, something that I think people would find useful would be to further define the parameters before a download. IE., I have to re-encode everything I am going to archive. I personally use cbr26 @ 96kbs, which works for me. The ability to set the bitrates and sizing for video / audio would save a lot of time. Example: College football I re-encode to 30cbr @ 96kbs x 576p.

Check out the beta I just put out and see what those ffmpeg options do for you:
https://github.com/jessedp/tablo-tools-electron/releases/tag/v0.3.13-beta.0

**Reply by JUser69** [reply]
Well, tried it out a couple of weeks ago and it crashed when I selected the new features.  Rolled back to latest version, have not been able to test lately because my tablo unit is down,  Getting them to issue an RMA is like pulling teeth.  You would think it was going to bankrupt the company...

**Comment by dougK44883** [comment]
If this is crucially important and want to do some manual intervention, actually scripting makes it easier, you can use `ffmpeg` directly on the playlist.  Quick overview - using Tablo Tools find an episode via Search or Browse and click on the "i" icon.  There's a green URL with your tablo's IP, click it.  In the directory listing click `pl/` [playlist].  I've always used `playlist.m3u8`.  From a now since gone github tool use this... if you know ffmpeg tweak it as needed.
`ffmpeg -loglevel warning -i 'http://tabloIP:8080/pvr/XXXXXX/pl/playlist.m3u8' -bsf:a aac_adtstoasc -vcodec copy -acodec copy -crf 18 "${2}".mp4`

I believe Movies and Sports are organized a bit different than TV show episodes, not sure.  I use to use this with a since unsupported 3rd party app - capto (and wrapto).

**Reply by JUser69** [reply]
Nice to know info.  Works in a pinch.  Still would be nice to have options built in, but I can wait....
